### PR TITLE
[VideoInfoScanner] Honor importwatchedstate and importresumepoint from AS.xml

### LINKED ISSUE #24909: importwatchedstate ignored in advancedsettings.xml from Kodi 20.0
### Describe the bug

My collection consist of movies with local .nfo files. I use it shared between multiple devices, both: Windows PCs and Raspberry Pis with libreelec.

My media manger allows me to mark movies as watched, this puts \<playcount\>1\</playcount\> in the .nfo file.
Back in Kodi 19.5 if i wanted the scanner to pick them up as watched i had to add advancedsettings.xml with importwatchedstate set to true.

This is no longer needed in Kodi 20. The scanner will always pick them up as watched if playcount is 1.

The problem thou is now i want a new install on a Raspberry Pi to have everything as unwatched. But adding the advancedsettings.xml with importwatchedstate set to false won't work.

I have also confirmed this on one of my windows installs that had importwatchedstate set to true, by changing it to false. And then renaming a movie with \<playcount\>1\</playcount\> but it still gets picked up by the scanner as watched.

If i change playcount to 0 and rename the movie again, then it will be picked up by the scanner as unwatched.

I have zero explanation for this behavior, curiously i had a look at the source AdvancedSettings.cpp and VideoInfoScanner.cpp, and it looks like it should work. But without being able to add more traces to the log and recompile i can't figure out the problem.

## Expected Behavior

With importwatchedstate set to false in advancedsettings.xml i expect the movie to be added by the scanner as unwatched even with playcount set to 1.

## Actual Behavior

Movie is always added as watched regardless of importwatchedstate set to true or false.

## Possible Fix

Go back to Kodi 19.5 or not use the playcount in the .nfo.

### To Reproduce

1. Install Kodi 20.0+ (i have tested with 20.0, 20.5 and 21.0-RC2)
2. Setup a Movie source with Local Information Only
3. Add a Movie together with .nfo file with \<playcount\>1\</playcount\>
4. Update library in Kodi, expect it to be picked up by scanner as watched.
5. Create advancedsettings.xml in userdata folder with:
```
<advancedsettings version="1.0">
    <videolibrary>
        <importwatchedstate>false</importwatchedstate>
        <importresumepoint>false</importresumepoint>
    </videolibrary>
</advancedsettings>
```
6. Rename Movie so it will be picked up by scanner again.
7. Restart Kodi so it loads advancedsettings.xml
8. Update library again, and it still gets picked up as watched

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/zevonexuqe.kodi

## Additional context or screenshots (if appropriate)

I think this thread might describe the same problem:
https://forum.kodi.tv/showthread.php?tid=375066

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [x] Windows
 - [x] Windows UWP

 - Operating system version/name: Windows 10, Windows 11, Libreelec
 - Kodi version: 20.0, 20.5, 21.0-rc2

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by neo1973**
## Description
When `importwatchedstate` and/or `importresumepoint` are set to `false` in `advancedsettings.xml` they should not be imported from NFOs.

## Motivation and context

Fixes #24909.

## How has this been tested?
Using this dummy NFO:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<movie>
  <title>Die Hard</title>
  <originaltitle>Die Hard</originaltitle>
  <year>1988</year>
  <playcount>1</playcount>
  <resume>
    <position>15</position>
    <total>100</total>
  </resume>
</movie>
```

And then testing all combinations of `importwatchedstate` and `importresumepoint`.

## What is the effect on users?

Users can again choose to not import watched state and resume point from NFOs.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/VideoInfoScanner.cpp [hunk 1]
```diff
@@ -744,7 +745,15 @@ namespace VIDEO
       if (loader)
       {
         pItem->GetVideoInfoTag()->Reset();
-        result = loader->Load(*pItem->GetVideoInfoTag(), false);
+        CVideoInfoTag& infoTag = *pItem->GetVideoInfoTag();
+        result = loader->Load(infoTag, false);
+
+        // only import some properties if advancedsettings.xml says so
```
Comment (CrystalP) [inline xbmc/video/VideoInfoScanner.cpp:751]: nitpick, maybe "keep some properties only if advancedsettings.xml says so" would be clearer

**Review by CrystalP — APPROVED** [review]
LGTM

File: xbmc/video/VideoInfoScanner.cpp (hunk 1 above)
Comment (neo1973) [reply xbmc/video/VideoInfoScanner.cpp]: Thanks, updated :+1:

**Comment by Vislike** [comment]
Since i am curious about the problem i checked your commit. Your fix is in a function called "RetrieveInfoForMovie", do you think the same fix need to be applied to the other functions as well for this to also work for tv shows and so on? I have not looked into this in details, was just a thought.

**Comment by neo1973** [comment]
Good point, as is this only works for movies. I'll have to look into a better solution to cover other media types as well.

**Comment by neo1973** [comment]
Jenkins build this please.

File: xbmc/video/VideoInfoScanner.cpp [hunk 2]
```diff
@@ -1964,14 +1966,7 @@ namespace KODI::VIDEO
       const ScraperPtr& info(scraper);
       std::unique_ptr<IVideoInfoTagLoader> loader;
       if (useLocal)
-      {
-        loader.reset(CVideoInfoTagLoaderFactory::CreateLoader(item, info, false));
-        if (loader)
-        {
-          // no reset here on purpose
```
Comment (CrystalP) [inline xbmc/video/VideoInfoScanner.cpp:1971]: Looks like this instance does not reset the InfoTag, unlike the other callers but ReadInfoTag() always resets it.

File: xbmc/video/VideoInfoScanner.cpp [hunk 3]
```diff
@@ -1688,6 +1668,28 @@ namespace KODI::VIDEO
...
+      CVideoInfoTag& infoTag = *item.GetVideoInfoTag();
+      infoTag.Reset();
+      result = loader->Load(infoTag, false);
+
+      // keep some properties only if advancedsettings.xml says so
+      const auto advancedSettings = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings();
+      if (!advancedSettings->m_bVideoLibraryImportWatchedState)
+        infoTag.ResetPlayCount();
+      if (!advancedSettings->m_bVideoLibraryImportResumePoint)
+        infoTag.SetResumePoint(CBookmark());
```
Comment (CrystalP) [inline xbmc/video/VideoInfoScanner.cpp:1688]: may not make much difference, the advanced settings could be retrieved once in CVideoInfoScanner()?

**Review by CrystalP — CHANGES_REQUESTED** [review]

File: xbmc/video/VideoInfoScanner.cpp (hunk 3 above)
Comment (neo1973) [reply xbmc/video/VideoInfoScanner.cpp]: OK, I did that in a separate commit which I won't include in the backport.

**Review by CrystalP — APPROVED** [review]
LGTM

**Pull request merged**